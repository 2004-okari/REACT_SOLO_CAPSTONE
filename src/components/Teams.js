import React from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import Navigation from './Navigation';
import './teams.css';

function Teams() {
  const teams = useSelector((state) => state.teams.teams.teams);
  const loading = useSelector((state) => state.teams.loading);
  const error = useSelector((state) => state.teams.error);

  let content;

  if (loading) {
    content = (
      <div className="loader">
        <div data-glitch="Loading..." className="glitch">
          Loading...
        </div>
      </div>
    );
  } else if (error) {
    content = <div className="error">{error}</div>;
  } else if (teams) {
    content = (
      <div className="box">
        {teams.map((team) => (
          <div className="box-2" key={team.idTeam}>
            <div className="cover-div">
              <img className="image" src={team.strTeamFanart4} alt="logo" />
            </div>
            <div className="inner-div">
              <p className="text-1">{team.strTeam}</p>
              <p className="text-2">
                since
                {team.intFormedYear}
              </p>
            </div>
            <div className="icon-div">
              <Link className="link" to={`/players/${team.idTeam}`}>
                <svg
                  className="text-3"
                  xmlns="http://www.w3.org/2000/svg"
                  width="24"
                  height="24"
                  viewBox="0 0 24 24"
                >
                  <path
                    fill="currentColor"
                    d="M4.8 21.57L7.222 24L19.2 12L7.222 0L4.8 2.43L14.347 12z"
                  />
                </svg>
              </Link>
            </div>
          </div>
        ))}
      </div>
    );
  } else {
    content = null;
  }

  return (
    <div className="section">
      <Navigation className="header-text" />
      <div className="section-2">{content}</div>
    </div>
  );
}

export default Teams;
