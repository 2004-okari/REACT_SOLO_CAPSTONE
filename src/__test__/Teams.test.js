import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Teams from '../components/Teams';

const mockStore = configureMockStore([]);

describe('Teams component works okay', () => {
  test('Teams renders correctly', () => {
    const store = mockStore({
      teams: {
        teams: [],
      },
    });

    const tree = renderer
      .create(
        <Provider store={store}>
          <Teams />
        </Provider>,
      );

    expect(tree).toMatchSnapshot();
  });
});
