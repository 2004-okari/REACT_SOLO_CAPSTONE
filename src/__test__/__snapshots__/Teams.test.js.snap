// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teams component works okay Teams renders correctly 1`] = `
<div
  className="section"
>
  <h1
    className="header-text"
  >
    Premier League
  </h1>
  <div
    className="section-2"
  />
</div>
`;
